File: src/main/java/cn/zjiali/robot/main/websocket/RobotWebSocketListener.java
package cn.zjiali.robot.main.websocket;

import cn.zjiali.robot.factory.ServiceFactory;
import okhttp3.Response;
import okhttp3.WebSocket;
import okhttp3.WebSocketListener;
import okio.ByteString;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * @author zJiaLi
 * @since 2021-07-29 10:19
 */
public class RobotWebSocketListener extends WebSocketListener {

    private final WebSocketManager webSocketManager = ServiceFactory.getInstance().getBean(WebSocketManager.class.getSimpleName(), WebSocketManager.class);

    public RobotWebSocketListener() {
        super();
    }

    @Override
    public void onClosed(@NotNull WebSocket webSocket, int code, @NotNull String reason) {
        webSocketManager.removeSession("");
        super.onClosed(webSocket, code, reason);

    }

    @Override
    public void onClosing(@NotNull WebSocket webSocket, int code, @NotNull String reason) {
        super.onClosing(webSocket, code, reason);
    }

    @Override
    public void onFailure(@NotNull WebSocket webSocket, @NotNull Throwable t, @Nullable Response response) {
        super.onFailure(webSocket, t, response);
    }

    @Override
    public void onMessage(@NotNull WebSocket webSocket, @NotNull String text) {
        webSocketManager.handleMessage(webSocket, text);
        super.onMessage(webSocket, text);
    }

    @Override
    public void onMessage(@NotNull WebSocket webSocket, @NotNull ByteString bytes) {
        super.onMessage(webSocket, bytes);
    }

    @Override
    public void onOpen(@NotNull WebSocket webSocket, @NotNull Response response) {
        webSocketManager.putSession(webSocket);
        super.onOpen(webSocket, response);
    }
}

File: src/main/java/cn/zjiali/robot/main/websocket/RobotWebSocketListener.kt
package cn.zjiali.robot.main.websocket

import cn.zjiali.robot.util.GuiceUtil
import okhttp3.Response
import okhttp3.WebSocket
import okhttp3.WebSocketListener
import okio.ByteString

/**
 * @author zJiaLi
 * @since 2021-07-29 10:19
 */
class RobotWebSocketListener : WebSocketListener() {
    private val webSocketManager = GuiceUtil.getBean(
        WebSocketManager::class.java
    )

    override fun onClosed(webSocket: WebSocket, code: Int, reason: String) {
        webSocketManager.removeSession()
        super.onClosed(webSocket, code, reason)
    }

    override fun onClosing(webSocket: WebSocket, code: Int, reason: String) {
        webSocketManager.removeSession()
        super.onClosing(webSocket, code, reason)
    }

    override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) {
        webSocketManager.removeSession()
        super.onFailure(webSocket, t, response)
    }

    override fun onMessage(webSocket: WebSocket, text: String) {
        webSocketManager.handleMessage(webSocket, text)
        super.onMessage(webSocket, text)
    }

    override fun onMessage(webSocket: WebSocket, bytes: ByteString) {
        webSocketManager.handleMessage(webSocket, bytes.toString())
        super.onMessage(webSocket, bytes)
    }

    override fun onOpen(webSocket: WebSocket, response: Response) {
        webSocketManager.putSession(webSocket)
        super.onOpen(webSocket, response)
    }
}